R/lazyframe__lazy.R

```diff
@@ -510,39 +558,42 @@ LazyFrame_sink_parquet = function(
 #' # load ipc directly into a DataFrame / memory
 #' # pl$scan_ipc(tmpf2)$collect()
 LazyFrame_sink_ipc = function(
-    path,
-    compression = "zstd",
-    maintain_order = TRUE,
-    type_coercion = TRUE,
-    predicate_pushdown = TRUE,
-    projection_pushdown = TRUE,
-    simplify_expression = TRUE,
-    no_optimization = FALSE,
-    slice_pushdown = TRUE) {
+  path,
+  compression = "zstd",
+  maintain_order = TRUE,
+  type_coercion = TRUE,
+  predicate_pushdown = TRUE,
+  projection_pushdown = TRUE,
+  simplify_expression = TRUE,
+  slice_pushdown = TRUE,
+  no_optimization = FALSE,
+  inherit_optimization = FALSE
+) {
   if (isTRUE(no_optimization)) {
     predicate_pushdown = FALSE
     projection_pushdown = FALSE
     slice_pushdown = FALSE
   }
-
-  self |>
-    .pr$LazyFrame$optimization_toggle(
+  if (isTRUE(inherit_optimization)) {
+    self
+  } else {
+    self$set_optimization_toggle(
```

Comment:
any error in set_optimization_toggle will be raised and escaped the final unwrap

Reply:
consider using .pr$LazyFrame$set_opt... or wrap entire block in result() to catch any errors

Reply:
Fixed.

Reply:
thx I will review it soon